lib/executor/constant_arrival_rate_test.go

```diff
@@ -354,3 +354,41 @@ func TestConstantArrivalRateGlobalIters(t *testing.T) {
 		})
 	}
 }
+
+func TestConstantArrivalRateActiveVUs(t *testing.T) {
+	t.Parallel()
+
+	config := &ConstantArrivalRateConfig{
+		BaseConfig:      BaseConfig{GracefulStop: types.NullDurationFrom(0 * time.Second)},
+		TimeUnit:        types.NullDurationFrom(time.Second),
+		Rate:            null.IntFrom(10),
+		Duration:        types.NullDurationFrom(950 * time.Millisecond),
+		PreAllocatedVUs: null.IntFrom(5),
+		MaxVUs:          null.IntFrom(10),
+	}
+
+	var (
+		running          int64
+		getCurrActiveVUs func() int64
+		runMx            sync.Mutex
+	)
+
+	runner := simpleRunner(func(ctx context.Context, _ *lib.State) error {
+		runMx.Lock()
+		assert.Equal(t, atomic.AddInt64(&running, 1), getCurrActiveVUs())
```

Comment:
nit: either the atomic here is pointless (we already have mutex) or you should also use it t o read the value in the checks at the end of the test

Reply:
Good point. I removed the atomic in 7162f778.

The reason I didn't use it at the end is because the test is already done at that point, so it should be safe to read it directly.